File: app/src/main/java/com/supercasual/fourtop/uimain/topimages/TopImagesFragment.java
package com.supercasual.fourtop.uimain.topimages;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.supercasual.fourtop.databinding.FragmentTopImagesBinding;
import com.supercasual.fourtop.uimain.ImageAdapter;
import com.supercasual.fourtop.utils.Constants;

import org.jetbrains.annotations.NotNull;

public class TopImagesFragment extends Fragment {

    private FragmentTopImagesBinding binding;
    private TopImagesViewModel viewModel;
    private ImageAdapter adapter;


    @Override
    public View onCreateView(@NotNull LayoutInflater inflater,
                             ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentTopImagesBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        viewModel = new ViewModelProvider(this).get(TopImagesViewModel.class);

        viewModel.getApiTopImages().observe(getViewLifecycleOwner(), images -> {
            adapter.setImages(images);
        });

        LinearLayoutManager layoutManager = new LinearLayoutManager(requireContext());
        layoutManager.setReverseLayout(true);
        layoutManager.setStackFromEnd(true);
        binding.recyclerTopImages.setLayoutManager(layoutManager);
        adapter = new ImageAdapter();
        binding.recyclerTopImages.setAdapter(adapter);
        binding.recyclerTopImages.setHasFixedSize(true);

        setArguments();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        binding = null;
    }

    private void setArguments() {
        Bundle args = this.getArguments();

        if (args != null) {
            String token = args.getString(Constants.ARGS_TOKEN);

            if (token != null && !token.isEmpty()) {
                viewModel.getToken().setValue(token);
                viewModel.topPhotosRequest();
            } else {
                Toast.makeText(requireContext(), "Нет токена", Toast.LENGTH_SHORT).show();
            }
        }
    }
}

File: app/src/main/java/com/emikhalets/voteapp/view/topimages/TopImagesFragment.kt
package com.emikhalets.voteapp.view.topimages

import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import by.kirich1409.viewbindingdelegate.viewBinding
import com.emikhalets.voteapp.R
import com.emikhalets.voteapp.databinding.FragmentHomeBinding
import com.emikhalets.voteapp.test.createMockImages
import com.emikhalets.voteapp.utils.ACTIVITY
import com.emikhalets.voteapp.view.adapters.ImagesAdapter
import com.emikhalets.voteapp.view.base.SecondaryFragment
import timber.log.Timber

class TopImagesFragment : SecondaryFragment(R.layout.fragment_top_images) {

    private val binding: FragmentHomeBinding by viewBinding()
    private lateinit var imagesAdapter: ImagesAdapter
    private lateinit var llm: LinearLayoutManager

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        ACTIVITY.title = getString(R.string.top_images_title)
        initRecyclerView()
    }

    private fun initRecyclerView() {
        imagesAdapter = ImagesAdapter(true)
        Timber.d(imagesAdapter.toString())
        llm = LinearLayoutManager(this.context)
        binding.listImages.layoutManager = llm
        binding.listImages.setHasFixedSize(true)
        binding.listImages.isNestedScrollingEnabled = false
        binding.listImages.adapter = imagesAdapter
        imagesAdapter.updateList(createMockImages().sortedByDescending { it.rating })
    }
}